src/common.rs

```diff
@@ -188,3 +191,83 @@ pub struct Config {
     pub llvm_cxxflags: String,
     pub nodejs: Option<String>,
 }
+
+impl Config {
+
+    /// Add rustc flags to link with the crate's dependencies in addition to the crate itself
+    pub fn link_deps(&mut self) {
+        // The linked library path env var name depends on the target OS. I'm not certain if this
+        // covers all the cases adequately.
+        let (varname, path_delim) = if cfg!(any(target_os = "macos", target_os = "ios")) {
+            ("DYLD_LIBRARY_PATH", ":")
+        } else if cfg!(target_os = "windows") {
+            ("PATH", ";")
+        } else {
+            ("LD_LIBRARY_PATH", ":")
+        };
+
+        // Dependencies can be found in the environment variable. Throw everything there into the
+        // link flags
+        let lib_paths = env::var(varname).unwrap_or_else(|e| {
+            panic!("Cannot link to dependencies. Problem with env var '{}': {:?}", varname, e)
+        });
+
+        // Move the flags out so we can mutate them and put them back in
+        let flags_opt = mem::replace(&mut self.target_rustcflags, None);
+
+        // Append to current flags if any are set, otherwise make new String
+        let mut flags = flags_opt.unwrap_or(String::new());
+        for p in lib_paths.split(path_delim) {
```

Comment:
I think you can omit the `path_delim` completely and use `std::env::split_paths()`